OpenRA.Mods.Common/Traits/SelectionDecorations.cs

```diff
@@ -9,39 +9,77 @@
 #endregion
 
 using System.Collections.Generic;
+using System.Drawing;
 using System.Linq;
 using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Traits;
 
-namespace OpenRA.Traits
+namespace OpenRA.Mods.Common.Traits
 {
-	public class SelectionDecorationsInfo : ITraitInfo
+	public class SelectionDecorationsInfo : ITraitInfo, ISelectionDecorationsInfo
 	{
 		public readonly string Palette = "chrome";
+		[Desc("Visual bounds for selection box. If null, it uses AutoSelectionSize.")]
+		public readonly int[] VisualBounds = null;
+		[Desc("Health bar, production progress bar etc.")]
```

Comment:
Newlines are often helpful so you can see at first glance which Desc belongs to which fields.